### DISCUSSION TITLE
Does changing the contract resolver to camelCase break queries?

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#14763] [Q&A] [answered]
Let's say we have a simple query that looks something like this.

`session.Query<Workspace>().Where(x => x.Domain == "encom")`

When I change the contract resolver like this...

```
serializer.ContractResolver = new CamelCasePropertyNamesContractResolver();
```

the query fails.

Do I also need to add `PropertyName` attribute on all properties to make it work after changing the contract resolver? I haven't given this too much thought but it feels odd that Raven doesn't handle this by default.

**Comment by ayende** [comment, marked as answer]
The JSON serialization and the Linq support aren't plugged to the same thing.
We read the `JsonProperty` and `DataMember` attributes to see if you changed things, but we aren't plugged to the contract resolver.

Note that in the `Conventions` you have `PropertyNameConverter`, which is where you can plug that in.

**Reply by marcuslindblom** [reply]
Ok, so to be clear. If I change `PropertyNameConverter` like [this](https://ravendb.net/docs/article-page/5.4/csharp/client-api/configuration/conventions#using-camelcase-in-c-client) will make Query work without adding the `JsonProperty`?

**Reply by ayende** [reply]
Yes, that is the idea.

**Reply by marcuslindblom** [reply]
Ok, it seems like the autoindex still index things in PascalCase? I tried to delete the index but it is recreated in PascalCase.
<img width="1378" alt="Screenshot 2022-08-18 at 09 44 16" src="https://user-images.githubusercontent.com/319720/185338800-79375d77-a194-4293-b59a-61403c674b0d.png">

**Reply by ayende** [reply]
You are issuing the _query_ for pascal case, I think.
See the `ToString()` on the query

**Reply by marcuslindblom** [reply]
Ok, I don't really understand so I have sent a failing test to the support. I hope it helps.
